test/e2e/network/service.go

```diff
@@ -2321,7 +2321,7 @@ func execSourceipTest(pausePod v1.Pod, serviceAddress string) (string, string) {
 	timeout := 2 * time.Minute
 
 	e2elog.Logf("Waiting up to %v to get response from %s", timeout, serviceAddress)
-	cmd := fmt.Sprintf(`curl -q -s --connect-timeout 30 %s | grep client_address`, serviceAddress)
+	cmd := fmt.Sprintf(`curl -q -s --connect-timeout 30 %s/clientip`, serviceAddress)
```

Comment:
this is a very nice win potentially :-)

Reply:
`/clientip` part of the command line doesn't parse for me when looking at curl man pages.  Do you have a reference?

Reply:
yeah, that's because agnhost is a meta image that consolidated several images using subcommands. In this test it is using the `netexec` subcommand

https://github.com/kubernetes/kubernetes/pull/80595/files#diff-ed43f6fc75eb8ad508358ecf71f2b271R159

that starts a webserver with different endpoints:

https://github.com/kubernetes/kubernetes/blame/master/test/images/agnhost/README.md#L309